src/test/java/com/google/devtools/build/lib/remote/RemoteModuleTest.java

```diff
@@ -106,6 +115,266 @@ public final class RemoteModuleTest {
+        if (cacheCapabilities != null) {
+          assertThat(actionContextProvider.getRemoteCache()).isNotNull();
+          assertThat(actionContextProvider.getRemoteCache().getCacheCapabilities()).isNotNull();
+          assertThat(actionContextProvider.getRemoteCache().getCacheCapabilities())
+              .isEqualTo(cacheCapabilities);
+          grpcCacheClient = (GrpcCacheClient) actionContextProvider.getRemoteCache().cacheProtocol;
+          assertCircuitBreakerInstance(grpcCacheClient.getRetrier().getCircuitBreaker(), remoteOptions);
+        } else {
+          assertThat(actionContextProvider.getRemoteCache()).isNull();
+        }
+
+        if (hasRemoteExecutionCapability) {
+          assertThat(actionContextProvider.getRemoteExecutionClient()).isNotNull();
+          grpcRemoteExecutor = (GrpcRemoteExecutor) actionContextProvider.getRemoteExecutionClient();
+          assertCircuitBreakerInstance(grpcRemoteExecutor.getRetrier().getCircuitBreaker(), remoteOptions);
+        } else {
+          assertThat(actionContextProvider.getRemoteExecutionClient()).isNull();
+        }
+      }
+    } finally {
+      serverList.forEach(Server::shutdownNow);
+      for (Server server : serverList) {
+        server.awaitTermination();
+      }
+    }
+  }
+
+    @Test
+    public void testNetrc_netrcWithoutRemoteCache() throws Exception {
+        String netrc = "/.netrc";
+        FileSystem fileSystem = new InMemoryFileSystem(DigestHashFunction.SHA256);
+        Scratch scratch = new Scratch(fileSystem);
+        scratch.file(netrc, "machine foo.example.org login baruser password barpass");
+        AuthAndTLSOptions authAndTLSOptions = Options.getDefaults(AuthAndTLSOptions.class);
+        RemoteOptions remoteOptions = Options.getDefaults(RemoteOptions.class);
+
+        Cache<URI, ImmutableMap<String, ImmutableList<String>>> credentialCache =
+            Caffeine.newBuilder().build();
+
+        Credentials credentials =
+            RemoteModule.createCredentials(
+                CredentialHelperEnvironment.newBuilder()
+                    .setEventReporter(new Reporter(new EventBus()))
+                    .setWorkspacePath(fileSystem.getPath("/workspace"))
+                    .setClientEnvironment(ImmutableMap.of("NETRC", netrc))
+                    .setHelperExecutionTimeout(Duration.ZERO)
+                    .build(),
+                credentialCache,
+                new CommandLinePathFactory(fileSystem, ImmutableMap.of()),
+                fileSystem,
+                authAndTLSOptions,
+                remoteOptions);
+
+        assertThat(credentials).isNotNull();
+        assertThat(credentials.getRequestMetadata(URI.create("https://foo.example.org"))).isNotEmpty();
+        assertThat(credentials.getRequestMetadata(URI.create("https://bar.example.org"))).isEmpty();
+    }
+
+  private void assertCircuitBreakerInstance(Retrier.CircuitBreaker circuitBreaker, RemoteOptions remoteOptions) {
```

Comment:
indent

Can you take a look again at this change to see if you can rework it to not move so many methods around? Getting a handle on the depth of the change is difficult with all the sequencing changes.

Reply:
I tried to organize the test with no functional change and then added my test on top of it. There was lot of duplicate code. 

Let me know what you think about this.
1. I create another pr where I just reorganize the test.
2. Then I put this pr on top of that.

Reply:
I'm not sure what the indent level and position of testNetrc_netrcWithoutRemoteCache and similar methods could have to do with organization and duplicate code: the method appears to have been moved without any changes (sans indent).

If you think things will be made clearer with another PR, go for it.

Reply:
I didn't realize that this package uses 2 space indentation, while I was using 4 space indentation. Fix the indentation issue and made sure that the original position of the method are maintained.